sdg_mapping/sdg_index/sdg_index_utils.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+import pandas as pd
+
+import sys
+import os
+import json
+
+from sdg_mapping import project_dir
+
+with open(f'{project_dir}/data/aux/sdg_index_data_opts.json', 'r') as f:
+    state_dict = json.load(f)
+
+
+def sdg_index_file_path(year):
+    """sdg_index_file_path
+    To fetch filepath of SDG Index datasets.
+
+    Args:
+        year (str): Year of SDG Index datasets
+
+    Returns:
+        (str): File path
+
+    """
+    fname = f'{year}_sdg_index.xlsx'
+    return f'{project_dir}/data/raw/{fname}'
+
+
+def read_workbook(data_path, state):
+    """read_workbook
+    To read an Excel read_workbook
+
+    Args:
+        data_path (str): File path of an SDG Index dataset
+        state (str): State of dataset needed; raw or data
+
+    Returns:
+        (pd.DataFrame): Parsed SDG Index data
+
+    """
+    if state == 'data':
+        df = pd.read_excel(open(data_path, 'rb'),
+                  sheet_name=state_dict[state],header = 1)
+
+    elif state == 'raw':
+        df = pd.read_excel(open(data_path, 'rb'),
+                  sheet_name=state_dict[state])
+    return df
+
+def parse_2019_sdg_index(dataset, state):
+    """parse_2019_sdg_index
+    To clean SDG (2019) Index data
+
+    Args:
+        dataset (pd.DataFrame): SDG Index dataframe
+        state (str): State of dataset needed; raw or data
+
+    Returns:
+        (pd.DataFrame): Cleaned SDG Index data
+
+    """
+    with open(f'{project_dir}/data/aux/sdg_index_mappings.json', 'r') as f:
+        maps = json.load(f)
+
+    trend_map = maps['trend_map']
+    achievement_map = maps['achievement_map']
+
+    sdg_index_19_df = dataset.copy()
+    trend_columns = [i for i in sdg_index_19_df.columns if 'Trend' in i]
+    dashboard_columns = [i for i in sdg_index_19_df.columns if (('Dashboard' in i) and ('Goal' in i))]
+
+    # ignores '.' and replaces with NaN

```

Comment:
If `.` comes up a lot in the data as a null value then you could add it to the list of `na_values` used by pandas read_excel https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_excel.html